File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-3.java
class Solution {
    public int[] replaceElements(int[] arr) {
        int rightMax = -1;
        int[] res = new int[arr.length];
        for (int i = arr.length - 1; i > -1; i--){
            res[i] = rightMax;
            rightMax = Math.max(rightMax, arr[i]);
        }
        return res;
    }
}

File: Data Structures & Algorithms/sqrtx/submission-3.py
class Solution:
    def mySqrt(self, x: int) -> int:
        if x < 2:
            return x

        l, r = 0, x
        res = 0
        while l < r:
            m = l + (r - l) // 2
            if m * m <= x:
                res = m
                l = m + 1
            else:
                r = m
        return res